### DISCUSSION TITLE
Lsp for svelte

### DISCUSSION THREAD

**Discussion by bajrangCoder** [mkslanc/ace-linters#70] [Q&A] [answered]
Hi,
I have a question: Is it possible to add support for lsp in Svelte file by using 👉 https://github.com/sveltejs/language-tools ,

If yes then I will need to follow these same steps 👉 https://github.com/mkslanc/ace-linters/discussions/69#discussioncomment-6363721
Or anything else.

Mainly I want to add lsp for svelte file in this project.

**Comment by mkslanc** [comment, marked as answer]
Hi, @bajrangCoder. Yes you will need to follow those steps, with few differences. I guess [this example](https://github.com/sveltejs/svelte-atom/blob/master/lib/main.js#L109) would be more convenient in case you want to use through websockets. This [server spawning example](https://github.com/sveltejs/svelte-atom/blob/master/lib/main.js#L109) also could be helpful

**Comment by mkslanc** [comment]
And you don't need to implement it using BaseService and other stuff. This is separate client mode, which you could use to connect through `websocket` or `webworker`

**Comment by bajrangCoder** [comment]
Looks complicated, but I will try.
Thanks for answering

**Reply by mkslanc** [reply]
It would be less complicated, if there are any svelte language server wrapped by server on websockets, in that case you will need just start server on some port and register your service - like [here](https://github.com/mkslanc/ace-linters/blob/main/packages/demo/websockets-pylsp/client.ts#L9)

**Reply by bajrangCoder** [reply]
But I want to use it using web workers

**Comment by bajrangCoder** [comment]
Sir, i failed, Please you add this in this `ace-linters`

**Reply by mkslanc** [reply]
Seems I figured out how to do it with WebSockets (via web worker it would be really more complicated), but ace doesn't have svelte syntax highlighting. Could you create mode request in [ace main repository](https://github.com/ajaxorg/ace/issues)? I would use html one for now, but it doesn't have all necessary rules.

**Reply by bajrangCoder** [reply]
Ok, I already have a very basic implementation of svelte mode , here is that 👇
[mode-svelte.js](https://drive.google.com/file/d/1XefuKjSYzN8pQXtkoxjpBBh00nk1hrys/view?usp=drivesdk)

**Reply by legend-sabbir** [reply]
@mkslanc  
here is my implementation for svelte mode, 
any improvement i could make ?
https://dpaste.org/V7D2R

**Reply by mkslanc** [reply]
Either option could be a good starting point. Just create a PR for the [ace repo](https://github.com/ajaxorg/ace/pulls) and you can improve it over time (or other users can as well)."

**Reply by mkslanc** [reply]
btw: new version should support Svelte's WebSocket's Language Server. And I created [wrapper to start svelte language server](https://github.com/mkslanc/svelte-websockets-language-server). [Example of client](https://github.com/mkslanc/ace-linters/blob/main/packages/demo/websockets-sls/client.ts) . If you started server on 3030 port you could test it on https://mkslanc.github.io/ace-linters/svelte.html

**Reply by bajrangCoder** [reply]
Svelte language server doesn't works .
Here is the error:

```
svelte-ls error: Initialize language server at  /svelte-websockets-language-server

svelte-ls error: No svelte.config.js found. Using https://github.com/sveltejs/svelte-preprocess as fallback

svelte-ls error: Initialize new ts service at

svelte-ls error: Trying to load configs for /home/svelte-websockets-language-server

svelte-ls error: No svelte.config.js found. Using https://github.com/sveltejs/svelte-preprocess as fallback

svelte-ls error: SnapshotManager File Statistics:
Project files: 0
Svelte files: 0
From node_modules: 0
Total: 1
```

**Reply by mkslanc** [reply]
It's ok, they just pipe logs to error, there is no fatal errors now
![image](https://github.com/mkslanc/ace-linters/assets/32402726/76ff6317-e7ba-4616-8192-b0e2c04914c5)

**Reply by bajrangCoder** [reply]
Ok but still client don't works

**Reply by mkslanc** [reply]
how you are trying to connect? Is there are any errors in client's console?

**Reply by bajrangCoder** [reply]
No, but when i created a client on my local machine then it works , I don't know why 
But linting(error checking) doesn't works on local machine

**Reply by bajrangCoder** [reply]
![Screenshot_20230722-220531_1](https://github.com/mkslanc/ace-linters/assets/71929976/d33edd4c-a557-4bd1-b469-567845e48360)

**Reply by mkslanc** [reply]
It looks like version 0.12.0, I fixed it in 0.12.1

**Reply by bajrangCoder** [reply]
Hmm, 0.12.1 is working perfectly 👍
